src/Maestro/tests/Maestro.ScenarioTests/AsyncDisposableValue.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Threading.Tasks;
+
+namespace Maestro.ScenarioTests
+{
+    public class AsyncDisposableValue<T> : IAsyncDisposable
+    {
+        private readonly Func<ValueTask> _dispose;
+
+        public T Value { get; }
+
+        internal AsyncDisposableValue(T value, Func<ValueTask> dispose)
+        {
+            Value = value;
+            _dispose = dispose;
+        }
+
+        public ValueTask DisposeAsync()
+        {
+            return _dispose();
```

Comment:
Should we pass you the T? Presumably you might want it. :-) (And you can always ignore it)

Reply:
Done.